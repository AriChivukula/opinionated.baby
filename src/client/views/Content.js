// @flow

import React, { Component } from 'react';
import {
  Card,
  CardPrimaryAction,
  CardMedia,
  CardAction,
  CardActions,
  CardActionButtons,
  CardActionIcons
} from 'rmwc/Card';
import {
  Grid,
  GridCell,
  GridInner
} from 'rmwc/Grid';
import {
  List,
  SimpleListItem
} from 'rmwc/List';
import { Typography } from 'rmwc/Typography';

type Props = {
}

class Content extends Component<Props> {

  render() {
    return (
      <Grid>
        <GridCell span={12}>
          <Typography use="display2" tag="div">
            A confidently immature starting point
          </Typography>
        </GridCell>
        <GridCell span={12}>
          <Typography use="subheading2" tag="div">
            The hardest part of building a new dynamic website is avoiding
            all the dragons. Starting a development environment can freeze us
            with choice overload.
            {' '}
            <Typography
              theme="text-secondary-on-background"
              onClick={() => this.goto('https://github.com/arichiv/opinionated.baby/')}>
              Opinionated Baby
            </Typography>
            {' '}
            solves this by moving the starting line past the quagmire of setup
            and into the pleasure of production logic.
          </Typography>
        </GridCell>
        <GridCell span={12}>
          <Typography use="display2" tag="div">
            Important choices I made for you
          </Typography>
        </GridCell>
        <GridCell span={3}>
          <List twoLine>
            <SimpleListItem
              graphic="memory"
              text="Lambda"
              secondaryText="Server(less)"
              onClick={() => this.goto('https://aws.amazon.com/lambda/')}
            />
            <SimpleListItem
              graphic="share"
              text="PostgresSQL"
              secondaryText="Database"
              onClick={() => this.goto('https://www.postgresql.org/')}
            />
            <SimpleListItem
              graphic="storage"
              text="S3"
              secondaryText="Storage"
              onClick={() => this.goto('https://aws.amazon.com/s3/')}
            />
            <SimpleListItem
              graphic="bug_report"
              text="Travis CI"
              secondaryText="CI/CD"
              onClick={() => this.goto('https://travis-ci.org/')}
            />
          </List>
        </GridCell>
        <GridCell span={3}>
          <List twoLine>
            <SimpleListItem
              graphic="description"
              text="JavaScript"
              secondaryText="Language"
              onClick={() => this.goto('https://developer.mozilla.org/en-US/docs/Web/JavaScript/')}
            />
            <SimpleListItem
              graphic="language"
              text="Babel"
              secondaryText="Transpiler"
              onClick={() => this.goto('https://babeljs.io/')}
            />
            <SimpleListItem
              graphic="merge_type"
              text="Flow"
              secondaryText="Typing"
              onClick={() => this.goto('https://flow.org/')}
            />
            <SimpleListItem
              graphic="cached"
              text="Node.js"
              secondaryText="Runtime"
              onClick={() => this.goto('https://nodejs.org/en/')}
            />
          </List>
        </GridCell>
        <GridCell span={3}>
          <List twoLine>
            <SimpleListItem
              graphic="unarchive"
              text="yarn"
              secondaryText="Dependency"
              onClick={() => this.goto('https://yarnpkg.com/en/')}
            />
            <SimpleListItem
              graphic="build"
              text="Gulp"
              secondaryText="Toolchain"
              onClick={() => this.goto('https://gulpjs.com/')}
            />
            <SimpleListItem
              graphic="playlist_add_check"
              text="Jest"
              secondaryText="Testing"
              onClick={() => this.goto('https://facebook.github.io/jest/')}
            />
            <SimpleListItem
              graphic="graphic_eq"
              text="GraphQL"
              secondaryText="API"
              onClick={() => this.goto('https://graphql.org/')}
            />
          </List>
        </GridCell>
        <GridCell span={3}>
          <List twoLine>
            <SimpleListItem
              graphic="flip_to_front"
              text="React"
              secondaryText="Interface"
              onClick={() => this.goto('https://reactjs.org/')}
            />
            <SimpleListItem
              graphic="local_library"
              text="Relay"
              secondaryText="Fetch"
              onClick={() => this.goto('http://facebook.github.io/relay/')}
            />
            <SimpleListItem
              graphic="style"
              text="Material Design"
              secondaryText="Theme"
              onClick={() => this.goto('https://material.io/components/web/')}
            />
            <SimpleListItem
              graphic="line_style"
              text="SASS"
              secondaryText="Style"
              onClick={() => this.goto('https://sass-lang.com/')}
            />
          </List>
        </GridCell>
        <GridCell span={12}>
          <Typography use="display2" tag="div">
            <span style={{textDecoration: 'line-through'}}>
              Self justification
            </span>
            {' '}
            Philosophy
          </Typography>
        </GridCell>
        <GridCell span={12}>
          <Typography use="subheading2" tag="div">
            This framework is one-size-fits-me. It{'\''}s an open source
            version of a system I use when building complex websites.
            Anyone may attempt to use, contribute, or critique it. The
            consistency with which feedback is incorporated will be nothing
            short of capricious. My general mission and purpose is to support
            the vitality and happy, healthy development of our
            {' '}
            <Typography
              theme="text-secondary-on-background"
              onClick={() => this.goto('http://www.nic.baby/policies.html')}>
              babies and children
            </Typography>
            .
          </Typography>
        </GridCell>
        <GridCell span={12}>
          <Typography use="display2" tag="div">
            Release history
          </Typography>
        </GridCell>
        <GridCell span={4}>
          <Card>
            <CardPrimaryAction onClick={() => this.goto('https://github.com/arichiv/opinionated.baby/releases/tag/v1/')}>
              <CardMedia
                square
                style={{backgroundImage: 'url(images/v1.jpg)'}}
              />
              <Typography
                use="display1"
                style={{padding: '1rem'}}>
                <Typography theme="text-secondary-on-background">
                  v1
                </Typography>
                {' '}
                Dogged Dapper
              </Typography>
            </CardPrimaryAction>
          </Card>
        </GridCell>
        <GridCell span={4}>
          <Card>
            <CardPrimaryAction onClick={() => this.goto('https://github.com/arichiv/opinionated.baby/releases/tag/v0/')}>
              <CardMedia
                square
                style={{backgroundImage: 'url(images/v0.jpg)'}}
              />
              <Typography
                use="display1"
                style={{padding: '1rem'}}>
                <Typography theme="text-secondary-on-background">
                  v0
                </Typography>
                {' '}
                Cocksure Castle
              </Typography>
            </CardPrimaryAction>
          </Card>
        </GridCell>
      </Grid>
    );
  }

  goto(url: string): void {
    window.location = url;
  }
}

export default Content;
