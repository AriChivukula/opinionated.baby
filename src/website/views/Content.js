// @flow

import React, { Component } from 'react';
import {
  Card,
  CardPrimaryAction,
  CardMedia,
  CardAction,
  CardActions,
  CardActionButtons,
  CardActionIcons
} from 'rmwc/Card';
import {
  Grid,
  GridCell
} from 'rmwc/Grid';
import {
  List,
  SimpleListItem
} from 'rmwc/List';
import { Typography } from 'rmwc/Typography';

import { goto } from '../util.js';

type Props = {
}

class Content extends Component<Props> {

  render() {
    return (
      <Grid>
        <GridCell span={12}>
          <Typography use="display2" tag="div">
            A confidently immature starting point
          </Typography>
        </GridCell>
        <GridCell span={12}>
          <Typography use="subheading2" tag="div">
            The hardest part of building a new dynamic website is avoiding
            all the dragons. Starting a development environment can freeze us
            with choice overload.
            {' '}
            <Typography
              theme="text-secondary-on-background"
              onClick={goto('https://github.com/arichiv/opinionated.baby/')}>
              Opinionated Baby
            </Typography>
            {' '}
            solves this by moving the starting line past the quagmire of setup
            and into the pleasure of production logic.
          </Typography>
        </GridCell>
        <GridCell span={12}>
          <Typography use="display2" tag="div">
            Important choices I made for you
          </Typography>
        </GridCell>
        <GridCell span={3}>
          <List twoLine>
            <SimpleListItem
              graphic="memory"
              text="Lambda"
              secondaryText="Server(less)"
              onClick={goto('https://aws.amazon.com/lambda/')}
            />
            <SimpleListItem
              graphic="share"
              text="PostgreSQL"
              secondaryText="Database"
              onClick={goto('https://www.postgresql.org/')}
            />
            <SimpleListItem
              graphic="storage"
              text="S3"
              secondaryText="Storage"
              onClick={goto('https://aws.amazon.com/s3/')}
            />
            <SimpleListItem
              graphic="bug_report"
              text="Travis CI"
              secondaryText="CI/CD"
              onClick={goto('https://travis-ci.org/')}
            />
          </List>
        </GridCell>
        <GridCell span={3}>
          <List twoLine>
            <SimpleListItem
              graphic="description"
              text="JavaScript"
              secondaryText="Language"
              onClick={goto('https://developer.mozilla.org/en-US/docs/Web/JavaScript/')}
            />
            <SimpleListItem
              graphic="language"
              text="Babel"
              secondaryText="Transpiler"
              onClick={goto('https://babeljs.io/')}
            />
            <SimpleListItem
              graphic="cached"
              text="Node.js"
              secondaryText="Runtime"
              onClick={goto('https://nodejs.org/en/')}
            />
            <SimpleListItem
              graphic="desktop_mac"
              text="Electron"
              secondaryText="Desktop"
              onClick={goto('https://electronjs.org/')}
            />
          </List>
        </GridCell>
        <GridCell span={3}>
          <List twoLine>
            <SimpleListItem
              graphic="unarchive"
              text="yarn"
              secondaryText="Dependency"
              onClick={goto('https://yarnpkg.com/en/')}
            />
            <SimpleListItem
              graphic="build"
              text="Gulp"
              secondaryText="Toolchain"
              onClick={goto('https://gulpjs.com/')}
            />
            <SimpleListItem
              graphic="playlist_add_check"
              text="Jest"
              secondaryText="Testing"
              onClick={goto('https://facebook.github.io/jest/')}
            />
            <SimpleListItem
              graphic="graphic_eq"
              text="GraphQL"
              secondaryText="API"
              onClick={goto('https://graphql.org/')}
            />
          </List>
        </GridCell>
        <GridCell span={3}>
          <List twoLine>
            <SimpleListItem
              graphic="flip_to_front"
              text="React"
              secondaryText="Interface"
              onClick={goto('https://reactjs.org/')}
            />
            <SimpleListItem
              graphic="local_library"
              text="Relay"
              secondaryText="Fetch"
              onClick={goto('http://facebook.github.io/relay/')}
            />
            <SimpleListItem
              graphic="style"
              text="Material Design"
              secondaryText="Theme"
              onClick={goto('https://material.io/components/web/')}
            />
            <SimpleListItem
              graphic="line_style"
              text="SASS"
              secondaryText="Style"
              onClick={goto('https://sass-lang.com/')}
            />
          </List>
        </GridCell>
        <GridCell span={12}>
          <Typography use="display2" tag="div">
            <span style={{textDecoration: 'line-through'}}>
              Self justification
            </span>
            {' '}
            Philosophy
          </Typography>
        </GridCell>
        <GridCell span={12}>
          <Typography use="subheading2" tag="div">
            This framework is one-size-fits-me. It{'\''}s an open source
            version of a system I use when building complex websites.
            Anyone may attempt to use, critique, or contribute to it. The
            consistency with which feedback is incorporated will be nothing
            short of capricious. My general mission and purpose is to support
            the vitality and happy, healthy development of our
            {' '}
            <Typography
              theme="text-secondary-on-background"
              onClick={goto('http://www.nic.baby/policies.html')}>
              babies and children
            </Typography>
            .
          </Typography>
        </GridCell>
        <GridCell span={12}>
          <Typography use="display2" tag="div">
            Release history
          </Typography>
        </GridCell>
        <GridCell span={4}>
          <Card>
            <CardPrimaryAction onClick={goto('https://github.com/arichiv/opinionated.baby/releases/tag/v3/')}>
              <CardMedia
                square
                style={{backgroundImage: 'url(images/v3.jpg)'}}
              />
              <Typography
                use="display1"
                style={{padding: '1rem'}}>
                <Typography theme="text-secondary-on-background">
                  v3
                </Typography>
                <br />
                Tilted Turtle
              </Typography>
            </CardPrimaryAction>
          </Card>
        </GridCell>
        <GridCell span={4}>
          <Card>
            <CardPrimaryAction onClick={goto('https://github.com/arichiv/opinionated.baby/releases/tag/v2/')}>
              <CardMedia
                square
                style={{backgroundImage: 'url(images/v2.jpg)'}}
              />
              <Typography
                use="display1"
                style={{padding: '1rem'}}>
                <Typography theme="text-secondary-on-background">
                  v2
                </Typography>
                <br />
                Assertive Actor
              </Typography>
            </CardPrimaryAction>
          </Card>
        </GridCell>
        <GridCell span={4}>
          <Card>
            <CardPrimaryAction onClick={goto('https://github.com/arichiv/opinionated.baby/releases/tag/v1/')}>
              <CardMedia
                square
                style={{backgroundImage: 'url(images/v1.jpg)'}}
              />
              <Typography
                use="display1"
                style={{padding: '1rem'}}>
                <Typography theme="text-secondary-on-background">
                  v1
                </Typography>
                <br />
                Dogged Dapper
              </Typography>
            </CardPrimaryAction>
          </Card>
        </GridCell>
        <GridCell span={4}>
          <Card>
            <CardPrimaryAction onClick={goto('https://github.com/arichiv/opinionated.baby/releases/tag/v0/')}>
              <CardMedia
                square
                style={{backgroundImage: 'url(images/v0.jpg)'}}
              />
              <Typography
                use="display1"
                style={{padding: '1rem'}}>
                <Typography theme="text-secondary-on-background">
                  v0
                </Typography>
                <br />
                Cocksure Castle
              </Typography>
            </CardPrimaryAction>
          </Card>
        </GridCell>
      </Grid>
    );
  }
}

export default Content;
