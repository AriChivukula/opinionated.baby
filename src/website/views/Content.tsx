import * as React from "react";
import {
  Card,
  CardMedia,
  CardPrimaryAction,
  Grid,
  GridCell,
  List,
  SimpleListItem,
  Typography,
} from "rmwc";

import { goto } from "../util";

const fullWidth: number = 12;
const toolWidth: number = 3;
const releaseWidth: number = 4;

interface ITools {
  graphic: string;
  link: string;
  subtitle: string;
  title: string;
}

const tools: ITools[] = [
  {
    graphic: "memory",
    link: "https://aws.amazon.com/lambda/",
    subtitle: "Server(less)",
    title: "Lambda",
  },
  {
    graphic: "share",
    link: "https://www.postgresql.org/",
    subtitle: "Database",
    title: "PostgreSQL",
  },
  {
    graphic: "storage",
    link: "https://aws.amazon.com/s3/",
    subtitle: "Storage",
    title: "S3",
  },
  {
    graphic: "bug_report",
    link: "https://travis-ci.org/",
    subtitle: "CI/CD",
    title: "Travis CI",
  },
  {
    graphic: "language",
    link: "https://www.typescriptlang.org/",
    subtitle: "Language",
    title: "TypeScript",
  },
  {
    graphic: "description",
    link: "https://babeljs.io/",
    subtitle: "Transpiler",
    title: "Babel",
  },
  {
    graphic: "cached",
    link: "https://nodejs.org/en/",
    subtitle: "Runtime",
    title: "Node.js",
  },
  {
    graphic: "desktop_mac",
    link: "https://electronjs.org/",
    subtitle: "Desktop",
    title: "Electron",
  },
  {
    graphic: "unarchive",
    link: "https://yarnpkg.com/en/",
    subtitle: "Dependency",
    title: "yarn",
  },
  {
    graphic: "build",
    link: "https://gulpjs.com/",
    subtitle: "Toolchain",
    title: "Gulp",
  },
  {
    graphic: "playlist_add_check",
    link: "https://facebook.github.io/jest/",
    subtitle: "Testing",
    title: "Jest",
  },
  {
    graphic: "graphic_eq",
    link: "https://graphql.org/",
    subtitle: "API",
    title: "GraphQL",
  },
  {
    graphic: "flip_to_front",
    link: "https://reactjs.org/",
    subtitle: "Interface",
    title: "React",
  },
  {
    graphic: "local_library",
    link: "http://facebook.github.io/relay/",
    subtitle: "Fetch",
    title: "Relay",
  },
  {
    graphic: "style",
    link: "https://material.io/components/web/",
    subtitle: "Theme",
    title: "Material Design",
  },
  {
    graphic: "line_style",
    link: "https://sass-lang.com/",
    subtitle: "Style",
    title: "SASS",
  },
];

interface IReleases {
  titleA: string;
  titleB: string;
  version: string;
}

const releases: IReleases[] = [
  { titleA: "Radical", titleB: "Robot", version: "v6" },
  { titleA: "Fierce", titleB: "Feline", version: "v5" },
  { titleA: "Staunch", titleB: "Sibling", version: "v4" },
  { titleA: "Tilted", titleB: "Turtle", version: "v3" },
  { titleA: "Assertive", titleB: "Actor", version: "v2" },
  { titleA: "Dogged", titleB: "Dapper", version: "v1" },
  { titleA: "Cocksure", titleB: "Castle", version: "v0" },
];

export const Content: () => JSX.Element =
  (): JSX.Element => (
    <Grid>
      <GridCell span={fullWidth}>
        <Typography use="display2" tag="div">
          A confidently immature starting point
        </Typography>
      </GridCell>
      <GridCell span={fullWidth}>
        <Typography use="subheading2" tag="div">
          The hardest part of building a new dynamic website is avoiding
          all the dragons. Starting a development environment can freeze us
          with choice overload.
          {" "}
          <Typography
            theme="text-secondary-on-background"
            onClick={(): void => { goto("https://github.com/arichiv/opinionated.baby/"); }}>
            Opinionated Baby
          </Typography>
          {" "}
          solves this by moving the starting line past the quagmire of setup
          and into the pleasure of production logic.
        </Typography>
      </GridCell>
      <GridCell span={fullWidth}>
        <Typography use="display2" tag="div">
          Important choices I made for you
        </Typography>
      </GridCell>
      {tools.map((item: ITools, index: number) => (
        <GridCell span={toolWidth} key={index}>
          <List twoLine>
            <SimpleListItem
              graphic={item.graphic}
              text={item.title}
              secondaryText={item.subtitle}
              onClick={(): void => { goto(item.link); }}
            />
          </List>
        </GridCell>
      ))}
      <GridCell span={fullWidth}>
        <Typography use="display2" tag="div">
          <span style={{textDecoration: "line-through"}}>
            Self justification
          </span>
          {" "}
          Philosophy
        </Typography>
      </GridCell>
      <GridCell span={fullWidth}>
        <Typography use="subheading2" tag="div">
          This framework is one-size-fits-me. It's an open source
          version of a system I use when building complex websites.
          Anyone may attempt to use, critique, or contribute to it. The
          consistency with which feedback is incorporated will be nothing
          short of capricious. My general mission and purpose is to support
          the vitality and happy, healthy development of our
          {" "}
          <Typography
            theme="text-secondary-on-background"
            onClick={(): void => { goto("http://www.nic.baby/policies.html"); }}>
            babies and children
          </Typography>
          .
        </Typography>
      </GridCell>
      <GridCell span={fullWidth}>
        <Typography use="display2" tag="div">
          Release history
        </Typography>
      </GridCell>
      {releases.map((item: IReleases, index: number) => (
        <GridCell span={releaseWidth} key={index}>
          <Card>
            <CardPrimaryAction
              onClick={(): void => { goto(`https://github.com/arichiv/opinionated.baby/releases/tag/${item.version}/`); }}>
              <CardMedia
                square
                style={{backgroundImage: `url(/images/${item.version}.jpg)`}}
              />
              <Typography
                use="display1"
                style={{padding: "1rem"}}>
                <Typography theme="text-secondary-on-background">
                  {item.version}
                </Typography>
                <br />
                {item.titleA}
                <br />
                {item.titleB}
              </Typography>
            </CardPrimaryAction>
          </Card>
        </GridCell>
      ))}
    </Grid>
  );
