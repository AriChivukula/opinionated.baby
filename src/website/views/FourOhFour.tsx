/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<gen/website/views/FourOhFour.ts::module>>
 * BESPOKE<<FourOhFour>>
 * SIGNED<</uHsHlap12EM18ME0BsKqTXXjYq3pmJ2J2pcHrFT6XSrkC8m0cgtD7SpJ1iUh9i2wrgcIE6OcSOF+iSt7lDY1g==>>
 */

import * as React from "react";
import {
  Link,
} from "react-router-dom";
import {
  Typography,
} from "rmwc";

export function FourOhFour(
): JSX.Element {
  /* BESPOKE START <<FourOhFour>> */
  return (
    <Link to="/">
      <Typography use="display4" tag="div">
        FourOhFour
      </Typography>
    </Link>
  );
  /* BESPOKE END <<FourOhFour>> */
}
