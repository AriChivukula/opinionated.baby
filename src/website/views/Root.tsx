/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<gen/website/views.ts::RootReact>>
 * BESPOKE<<imports, render, implementation>>
 * SIGNED<<P1ff7Tq61T6TuGrXUQvDBChPYA3fkVQw5inLdp0dAapcC/WfbOxI0rMWNw2D4U6lQnQIs0JFh94DDLgp+a8NgA==>>
 */

import * as React from "react";

/* BESPOKE START <<imports>> */
import * as cookie from "js-cookie";

import {
  graphql,
  QueryRenderer,
} from "react-relay";
import {
  Environment,
} from "relay-runtime";

import {
  TopBarQuery,
} from "./__generated__/TopBarQuery.graphql";
import {
  Page,
} from "./Page";
/* BESPOKE END <<imports>> */

export interface IRootProps {
  environment: Environment;
}

class _Root extends React.Component<IRootProps> {

  public render(
  ): JSX.Element {
    /* BESPOKE START <<render>> */
    return (
      <QueryRenderer
        environment={this.props.environment}
        variables={{}}
        query={graphql`
          query RootQuery {
            ...TopBarQuery
          }
        `}
        render={({error, props}: {error: Error | null; props: TopBarQuery}): JSX.Element => {
          if (error !== null) {
            console.log(error);

            return <div />;
          } else {
            return <Page data={props} />;
          }
        }}
      />
    );
    /* BESPOKE END <<render>> */
  }

  /* BESPOKE START <<implementation>> */
  /* BESPOKE END <<implementation>> */
}

export { _Root as Root };
