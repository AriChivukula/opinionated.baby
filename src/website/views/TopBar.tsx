/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<gen/website/views.ts::TopBarReact>>
 * BESPOKE<<imports, render, implementation, relay>>
 * SIGNED<<cc1AN1qU5ZOxiGxyc9wF1QlFpobHngrvXZxS4gb4MbFdF9Kwsq6BXfxlyscRAxaEjJCZjKPpCBzJFES+sPkP3g==>>
 */

import * as React from "react";
import {
  polyfill,
} from "react-lifecycles-compat";
import {
  commitMutation,
  createFragmentContainer,
  graphql,
} from "react-relay";

/* BESPOKE START <<imports>> */
import * as cookie from "js-cookie";

import Relay from "relay-runtime";
import {
  Toolbar,
  ToolbarFixedAdjust,
  ToolbarIcon,
  ToolbarRow,
  ToolbarSection,
  ToolbarTitle,
} from "rmwc/Toolbar";
import {
  Url,
} from "url";

import {
  goto,
} from "../util";
import {
  TopBarLoginMutationResponse,
} from "./__generated__/TopBarLoginMutation.graphql";
import {
  TopBarLogoutMutationResponse,
} from "./__generated__/TopBarLogoutMutation.graphql";
import {
  TopBarQuery,
} from "./__generated__/TopBarQuery.graphql";
/* BESPOKE END <<imports>> */

export interface ITopBarProps {
  data: TopBarQuery;
  relay: Relay;
}

class __TopBar extends React.Component<ITopBarProps> {

  public render(
  ): JSX.Element {
    /* BESPOKE START <<render>> */
    const urlParts: URL = new URL(window.location.href);
    if (urlParts.searchParams.has("code")) {
      this.login(urlParts.searchParams.get("code") as string);

      return <div />;
    }
    let login: JSX.Element = (
      <ToolbarIcon
        use="person"
        onClick={(): void => { this.googleAuth(); }}
      />
    );
    if (this.props.data.me !== null) {
      login = (
        <>
          <ToolbarTitle>{this.props.data.me.email}</ToolbarTitle>
          <ToolbarIcon
            use="exit_to_app"
            onClick={(): void => { this.logout(); }}
          />
        </>
      );
    }

    return (
      <>
        <Toolbar fixed waterfall>
          <ToolbarRow>
            <ToolbarSection alignStart>
              <ToolbarTitle onClick={(): void => { goto("https://github.com/arichiv/opinionated.baby/"); }}>
                {document.title}
              </ToolbarTitle>
            </ToolbarSection>
            <ToolbarSection alignEnd>
              {login}
              <ToolbarIcon
                use="code"
                onClick={(): void => { goto("https://github.com/arichiv/opinionated.baby/"); }}
              />
              <ToolbarIcon
                use="info"
                onClick={(): void => { goto("http://chivuku.la/"); }}
              />
            </ToolbarSection>
          </ToolbarRow>
        </Toolbar>
        <ToolbarFixedAdjust />
      </>
    );
    /* BESPOKE END <<render>> */
  }

  /* BESPOKE START <<implementation>> */
  private googleAuth(
  ): void {
    if ("loginURL" in this.props.data) {
      goto(this.props.data.loginURL, true);
    }
  }

  private login(
    code: string,
  ): void {
    commitMutation(
      this.props.relay.environment,
      {
        mutation: graphql`
          mutation TopBarLoginMutation($input: LoginInput) {
            login(input: $input) {
              accessToken
            }
          }
        `,
        onCompleted: (response: TopBarLoginMutationResponse, errors: Error[]): void => {
          cookie.set(
            "accessToken",
            response.login.accessToken,
          );
          goto("/", true);
        },
        variables: {
          input: {
            code,
          },
        },
      },
    );
  }

  private logout(
  ): void {
    commitMutation(
      this.props.relay.environment,
      {
        mutation: graphql`
          mutation TopBarLogoutMutation($input: LogoutInput) {
            logout(input: $input) {
              accessToken
            }
          }
        `,
        onCompleted: (response: TopBarLogoutMutationResponse, errors: Error[]): void => {
          cookie.set(
            "accessToken",
            response.logout.accessToken,
          );
          goto("/", true);
        },
        variables: {
          input: {
            dummy: "",
          },
        },
      },
    );
  }
  /* BESPOKE END <<implementation>> */
}

const _TopBar: React.ComponentType = createFragmentContainer(
  __TopBar,
  /* BESPOKE START <<relay>> */
  graphql`
    fragment TopBarQuery on Query {
      loginURL,
      me {
        id,
        email
      }
    }
  `,
  /* BESPOKE END <<relay>> */
);

// @ts-lint
export const TopBar: React.Component<ITopBarProps> = polyfill(_TopBar);
