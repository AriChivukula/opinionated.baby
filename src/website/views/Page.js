// @flow

import type {RootQuery} from './__generated__/RootQuery.graphql.js';

import React, { Component, Fragment } from 'react';
import { LinearProgress } from 'rmwc/LinearProgress';
import url from 'url';

import Content from './Content.js';
import TopBar from './TopBar.js';

type Props = {
  data: ?RootQuery
}

class Page extends Component<Props> {

  render() {
    const url_parts = url.parse(window.location.href, true);
    if (!this.props.data || (url_parts.query && url_parts.query.code)) {
      return <LinearProgress determinate={false} />;
    } else {
      return (
        <Fragment>
          <TopBar data={this.props.data} />
          <Content />
        </Fragment>
      );
    }
  }
}

export default Page;
