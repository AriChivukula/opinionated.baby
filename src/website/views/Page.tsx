import { TopBarQuery } from "./__generated__/TopBarQuery.graphql";

import * as React from "react";
import { LinearProgress } from "rmwc";
import { parse, UrlWithParsedQuery } from "url";

import { Content } from "./Content";
import { TopBar } from "./TopBar";

interface IProps {
  data: TopBarQuery | undefined;
}

export class Page extends React.Component<IProps> {

  public render(): JSX.Element {
    const urlParts: UrlWithParsedQuery = parse(window.location.href, true);
    if (this.props.data === undefined || "code" in urlParts.query) {
      return <LinearProgress determinate={false} />;
    } else {
      return (
        <>
          <TopBar {...this.props} />
          <Content />
        </>
      );
    }
  }
}
