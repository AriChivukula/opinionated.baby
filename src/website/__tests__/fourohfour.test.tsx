import "jest-enzyme";
import "mutationobserver-shim";

import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";

import { FourOhFour } from "../views/FourOhFour";

Enzyme.configure({
  adapter: new Adapter(),
});

test(
  "FourOhFour",
  async (): Promise<void> => {
    expect(
      Enzyme.render(
        Enzyme.mount(
          <BrowserRouter>
            <FourOhFour />
          </BrowserRouter>,
        ),
      ),
    )
      .toMatchSnapshot();
  },
);
