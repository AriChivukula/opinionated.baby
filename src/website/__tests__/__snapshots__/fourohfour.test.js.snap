// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FourOhFour 1`] = `
<a
  href="/"
>
  <div
    class="mdc-typography--headline2"
  >
    FourOhFour
  </div>
</a>
`;
